library/entropy.c

```diff
@@ -466,6 +466,80 @@ static int entropy_dummy_source( void *data, unsigned char *output,
     return( 0 );
 }
 
+#if !defined(MBEDTLS_TEST_NULL_ENTROPY) && defined(MBEDTLS_ENTROPY_HARDWARE_ALT)
+
+static int entropy_self_test_bias_gather( unsigned char *buf, size_t buf_len )
+{
+    int ret = 0;
+    size_t entropy_len = 0;
+    size_t olen;
```

Comment:
`olen` is passed uninitialised and then used for addition. Safer to initialise it here to 0.